pkg/curl/schemes.go

```diff
@@ -395,7 +396,7 @@ func httpFetch(ctx context.Context, c *http.Client, u *url.URL) (io.Reader, erro
 	}
 
 	if resp.StatusCode != 200 {
-		return nil, &HTTPClientCodeError{err, resp.StatusCode}
+		return nil, &HTTPClientCodeError{ErrStatusNotOk, resp.StatusCode}
```

Comment:
is this change needed? Isn't `ErrStatusNotOk` redundant with `resp.StatusCode != 200`, which can be determined by the caller of this function?  This changes the behavior such that the returned `HTTPClientCodeErr.Err` field is no longer `nil` which may not be backwards compatible - are we sure this is safe?

Reply:
it's needed if we want to use `errors.Is`. Our curl package treat any non 200 response as an error. And because err in `&HTTPClientCodeError{err, resp.StatusCode}` is nil we can't unwrap it to an error.

It's not clear to me why any non 200 code is an error, but it's kinda different story.